cli/cli/kurtosis_config/resolved_config/kurtosis_cluster_config_test.go

```diff
@@ -252,19 +265,113 @@ func TestNewKurtosisClusterConfigShouldEnableDefaultLogsSink(t *testing.T) {
+		EnclaveSizeInMegabytes: &kubernetesEnclaveSizeInMB,
+		EngineNodeName:         &kubernetesEngineNodeName,
+	}
+	kurtosisClusterConfigOverrides := v6.KurtosisClusterConfigV6{
+		Type:                        &kubernetesType,
+		Config:                      &kubernetesFullConfig,
+		LogsAggregator:              nil,
+		LogsCollector:               nil,
+		GrafanaLokiConfig:           nil,
+		ShouldEnableDefaultLogsSink: nil,
+	}
+	actualKurtosisClusterConfig, err := NewKurtosisClusterConfigFromOverrides("test", &kurtosisClusterConfigOverrides)
+	require.NoError(t, err)
+	require.Nil(t, actualKurtosisClusterConfig.logsCollector.Filters)
+	require.Nil(t, actualKurtosisClusterConfig.logsCollector.Parsers)
+}
+
+func TestNewKurtosisClusterConfigLogsCollectorFullConfig(t *testing.T) {
+	kubernetesType := KurtosisClusterType_Kubernetes.String()
+	kubernetesClusterName := "some-name"
+	kubernetesStorageClass := "some-storage-class"
+	kubernetesEnclaveSizeInMB := uint(5)
+	kubernetesEngineNodeName := "some-node-name"
+	kubernetesFullConfig := v6.KubernetesClusterConfigV6{
+		KubernetesClusterName:  &kubernetesClusterName,
+		StorageClass:           &kubernetesStorageClass,
+		EnclaveSizeInMegabytes: &kubernetesEnclaveSizeInMB,
+		EngineNodeName:         &kubernetesEngineNodeName,
+	}
+	kurtosisClusterConfigOverrides := v6.KurtosisClusterConfigV6{
+		Type:   &kubernetesType,
+		Config: &kubernetesFullConfig,
+		LogsAggregator: &v6.LogsAggregatorConfigV6{
+			Sinks: map[string]map[string]interface{}{
+				"elasticsearch": {
+					"type": "elasticsearch",
+				},
+			},
+		},
+		LogsCollector: &v6.LogsCollectorConfigV6{
+			Filters: []logs_collector.Filter{
+				{
+					Name:  "grep",
+					Match: "*",
+					Params: []logs_collector.FilterParam{
+						{Key: "exclude", Value: "*"},
+						{Key: "logical_op", Value: "&"},
+					},
+				},
+				{
+					Name:  "lua",
+					Match: "*",
+					Params: []logs_collector.FilterParam{
+						{Key: "script", Value: "print smth"},
+						{Key: "call", Value: "frontend"},
+					},
+				},
+			},
+			Parsers: []logs_collector.Parser{
+				{
+					"name":        "json",
+					"format":      "json",
+					"time_key":    "time",
+					"time_format": "%Y-%m-%dT%H:%M:%S.%L",
+				},
+				{
+					"name":   "regex",
+					"format": "regex",
+					"regex":  "^\\[(?<time>[^\\]]*)\\] (?<level>\\w+) (?<message>.*)$",
+				},
+			},
+		},
+		GrafanaLokiConfig:           nil,
+		ShouldEnableDefaultLogsSink: nil,
+	}
+	actualKurtosisClusterConfig, err := NewKurtosisClusterConfigFromOverrides("test", &kurtosisClusterConfigOverrides)
+	require.NoError(t, err)
+	require.NotNil(t, actualKurtosisClusterConfig.logsCollector)
+	require.NotNil(t, actualKurtosisClusterConfig.logsCollector.Filters)
+	require.NotNil(t, actualKurtosisClusterConfig.logsCollector.Parsers)
+	require.Equal(t, 2, len(actualKurtosisClusterConfig.logsCollector.Filters))
+	require.Equal(t, 2, len(actualKurtosisClusterConfig.logsCollector.Parsers))
```

Comment:
Up to you, but maybe worth verifying that the values you pumped in are getting returned (since last PR we had a bug where values weren't coming out)